openslides_backend/action/actions/user/save_saml_account.py

```diff
@@ -465,10 +459,26 @@ def get_field_data(
                             )
                         )
                     elif saml_meeting_user_field == "vote_weight":
-                        # Result is string and has 6 digits.
-                        if isinstance(value, str):
-                            value = float(value)
-                        result = f"{value:f}"
+                        # Result must be string and have 6 digits after dot.
+                        if isinstance(value, int):
+                            value = f"{value:f}"
+                        elif isinstance(value, str) and re.compile(
+                            r"^[1-9]\d*\.?\d{0,6}$"
+                        ).match(value):
+                            if re.compile(r"^\d*$").match(value):
+                                value += ".000000"
+                            else:
+                                regex_condition = re.compile(DECIMAL_REGEX)
```

Comment:
`DECIMAL_REGEX` is already compiled, so no need to do it again ;-)

Reply:
I used the 'DECIMAL_PATTERN' now.